### PR TITLE
Smallish stream bits

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

File: akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala [hunk 1]
```diff
@@ -558,6 +561,9 @@ object TestSubscriber {
 
     private lazy val subscription = expectSubscription()
 
+    /** Asserts that a subscription has been received or will be received */
+    def ensureSubscription(): Unit = subscription // initializes lazy val
```
Comment (ktoso) [inline akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala]: oh very interesting catch...

**Comment by ktoso** [comment]
LGTM

File: akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala (hunk 1 above)
Comment (drewhk) [inline akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala]: I run into this before and worked around, but this is nicer.

File: akka-stream/src/main/scala/akka/stream/scaladsl/BidiFlow.scala [hunk 2]
```diff
@@ -117,6 +117,9 @@ final class BidiFlow[-I1, +O1, -I2, +O2, +Mat](private[stream] override val modu
    */
   def reversed: BidiFlow[I2, O2, I1, O1, Mat] = new BidiFlow(module.replaceShape(shape.reversed))
 
+  def mapMaterializedValue[Mat2](f: Mat ⇒ Mat2): BidiFlow[I1, O1, I2, O2, Mat2] =
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/scaladsl/BidiFlow.scala]: thanks. Does it exist for Java side?

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/scaladsl/BidiFlow.scala]: good catch @drewhk, it's missing.

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/scaladsl/BidiFlow.scala]: @jrudolph do you have a sec to add to `javadsl.BidiFlow`?

Comment (jrudolph) [inline akka-stream/src/main/scala/akka/stream/scaladsl/BidiFlow.scala]: Yes, I'll give it a try.

Comment (jrudolph) [inline akka-stream/src/main/scala/akka/stream/scaladsl/BidiFlow.scala]: Updated.

**Comment by ktoso** [comment]
Thanks @jrudolph! Will merge after validation.

**Pull request merged**